# HG changeset patch
# User Rik <rik@octave.org>
# Date 1383454919 25200
#      Sat Nov 02 22:01:59 2013 -0700
# Node ID cfe5297e0f3b1d2699277d9604898c67a6ddc99a
# Parent  73b239db159448f515c34817606e57fadaa6ecb9
figure.m: Recode to pass failing test on Mac OS X.

* scripts/plot/util/figure.m: Require that input is a scalar.

diff --git a/scripts/plot/util/figure.m b/scripts/plot/util/figure.m
--- a/scripts/plot/util/figure.m
+++ b/scripts/plot/util/figure.m
@@ -48,22 +48,22 @@ function h = figure (varargin)
   if (mod (nargs, 2) == 0)
     f = NaN;
     init_new_figure = true;
   else
     arg = varargin{1};
     if (ischar (arg))
       arg = str2double (arg);
     endif
-    if (isfigure (arg))
+    if (isscalar (arg) && isfigure (arg))
       f = arg;
       init_new_figure = false;
       varargin(1) = [];
       nargs--;
-    elseif (isnumeric (arg) && isscalar (arg) && arg > 0 && arg == fix (arg))
+    elseif (isscalar (arg) && isnumeric (arg) && arg > 0 && arg == fix (arg))
       f = arg;
       init_new_figure = true;
       varargin(1) = [];
       nargs--;
     else
       error ("figure: N must be figure handle or figure number");
     endif
   endif
